pallets/msa/src/lib.rs

```diff
@@ -81,10 +82,26 @@ pub mod pallet {
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		MsaCreated { msa_id: MessageSenderId, key: T::AccountId },
-		KeyAdded { msa_id: MessageSenderId, key: T::AccountId },
-		KeyRevoked { key: T::AccountId },
-		DelegateAdded { delegate: Delegate, delegator: Delegator },
+		MsaCreated {
+			msa_id: MessageSenderId,
+			key: T::AccountId,
+		},
+		KeyAdded {
+			msa_id: MessageSenderId,
+			key: T::AccountId,
+		},
+		KeyRevoked {
+			key: T::AccountId,
+		},
+		DelegateAdded {
+			delegate: Delegate,
+			delegator: Delegator,
+		},
+		MsaCreatedAndDelegated {
```

Comment:
Let me know if this makes sense or maybe we might want to have something else as an event for this action

Reply:
I wonder if we should separate these into two events since we're also allowing adding delegates, and also account creation via token payment.

Reply:
I like that idea.

Reply:
OK, I can modify it to add two events but then it would be harder to know if this was created using delegation or not